design/Events_design.md

```diff
@@ -0,0 +1,314 @@
+    /// @param tickSpacing The minimum number of ticks between initialized ticks
+    /// @param pool The address of the created pool
+    event PoolCreated(
+        address indexed token0,
+        address indexed token1,
+        uint24 indexed fee,
+        int24 tickSpacing,
+        address pool
+    );
+```
+
+```
+    /// @notice Emitted when a new fee amount is enabled for pool creation via the factory
+    /// @param fee The enabled fee, denominated in hundredths of a bip
+    /// @param tickSpacing The minimum number of ticks between initialized ticks for pools created with the given fee
+    event FeeAmountEnabled(uint24 indexed fee, int24 indexed tickSpacing);
+```
+
+```
+    event Sync(uint112 reserve0, uint112 reserve1);
+```
+
+```
+    /// @notice Emitted when the protocol fee is changed by the pool
+    /// @param feeProtocol0Old The previous value of the token0 protocol fee
+    /// @param feeProtocol1Old The previous value of the token1 protocol fee
+    /// @param feeProtocol0New The updated value of the token0 protocol fee
+    /// @param feeProtocol1New The updated value of the token1 protocol fee
+    event SetFeeProtocol(uint8 feeProtocol0Old, uint8 feeProtocol1Old, uint8 feeProtocol0New, uint8 feeProtocol1New);
+```
+
+What these events have in common is that they are not user-specific. They represent a general update of the smart contract.
+
+##### Visibility Rules
+
+Users should be able to request and read all events that are relevant to them.
+
+The rules we propose are:
+
+1. An event is considered relevant to all account owners whose addresses are used as topics in the event.
+2. In case there is no address in a topic, then the event is considered a lifecycle event, and thus relevant to everyone.
+3. An event is considered relevant to the signer of the transaction that created it. 
+
+The purpose for these rules is to be simple, clear, intuitive, and to work as good as possible with the existing contracts.
+
+The first two rules are intuitive. The third one adds another dimension to the reasoning process, because there is an implicit user to whom the event is relevant.
+The alternative is to remove the third rule, and allow the owner to view the event only if it contains its own address. 
+
+*Note: If we decide to remove the third rule, the implementation will have to calculate the event visibility when adding them to transaction receipts.*
+
+In case the rules above are not providing the desired functionality, the developer can adjust the topics.
+
+For example, if one of the lifecycle events should only be visible to the administrators, the developer can add that address as a topic.
+Note that the lifecycle event will be visible to the signer of the transaction as well (according to rule 3).
+
+In case an event with an address field that should not contribute to relevancy, the developer can remove the `"indexed"` and thus the event will become invisible to that user.
```

Comment:
Can this be expanded. In this scenario what does contribute to relevancy mean?

Reply:
being an indexed field (topic) in that log entry ( event)